FLT/DedekindDomain/FiniteAdeleRing/IsDirectLimit.lean

```diff
@@ -0,0 +1,151 @@
+/-
+Copyright (c) 2025 Madison Crim. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Madison Crim
+-/
+
+import Mathlib.Algebra.Colimit.Module
+
+section IsDirectLimit
+
+variable {R : Type*} [Semiring R] {ι : Type*} [Preorder ι] (M : ι → Type*) (P : Type*)
+ [AddCommMonoid P] [Module R P] [∀ i, AddCommMonoid (M i)] [∀ i, Module R (M i)]
+ (f : ∀ i j, i ≤ j → M i →ₗ[R] M j) (g : ∀ i, M i →ₗ[R] P)
+
+/--
+`IsDirectLimit f g` states that `P` is the direct limit of the system `(M i, f i j hij)`.
+It requires:
+* `surj`: says every element of `P` comes from some `M i` via the canonical map `g i`.
+* `inj`: says that if two elements of (possibly different) modules `M i` and `M j` map
+  to the same element of `P`, then they become equal after being mapped into a common
+  later module `M k`.
+* `compatibility`: ensures that the maps `g i` and `f i j hij` are compatible with one another.
+-/
+@[mk_iff] class IsDirectLimit : Prop where
```

Comment:
Also I think we should state this in terms of the underlying function because the direct limit is the same as the direct limit of the underlying sets, and `DirectedSystem` is also stated in terms of unbundled functions.
```
variable {ι : Type*} [Preorder ι] (M : ι → Type*) (P : Type*)
 (f : ∀ i j, i ≤ j → M i → M j) (g : ∀ i, M i → P) in
@[mk_iff] class IsDirectLimit [DirectedSystem M f] : Prop where
  surj : ∀ m : P, ∃ i, ∃ mi : M i, g i mi = m
  inj :  ∀ i j, ∀ mi : M i, ∀ mj : M j, g i mi = g j mj → ∃ (k : ι) (hik : i ≤ k) (hjk : j ≤ k),
      f i k hik mi = f j k hjk mj
  compatibility : ∀ i j hij x, g j (f i j hij x) = g i x
```